src/engine/renderer/tr_shader.cpp

```diff
@@ -7155,11 +7177,41 @@ static void CreateInternalShaders()
 	Q_strncpyz( shader.name, "<default>", sizeof( shader.name ) );
 
 	shader.type = shaderType_t::SHADER_3D_DYNAMIC;
+	shader.noFog = true;
+	shader.fogShader = nullptr;
 	stages[ 0 ].type = stageType_t::ST_DIFFUSEMAP;
 	stages[ 0 ].bundle[ 0 ].image[ 0 ] = tr.defaultImage;
 	stages[ 0 ].active = true;
 	stages[ 0 ].stateBits = GLS_DEFAULT;
 	tr.defaultShader = FinishShader();
+
+	Q_strncpyz( shader.name, "<fogEqual>", sizeof( shader.name ) );
+
+	shader.type = shaderType_t::SHADER_3D_DYNAMIC;
+	shader.sort = Util::ordinal( shaderSort_t::SS_FOG );
+	shader.noFog = true;
```

Comment:
I don't see why we should specially disable fog for the not-found shader.

Reply:
I suppose that line can be dropped.

Reply:
I tried it, it seems to work fine, so I added that change.